src/communication/RpcServer.cpp

```diff
@@ -10,3 +10,125 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/communication/RpcServer.h"
+
+namespace uprotocol::communication {
+
+namespace Validator = datamodel::validator;
+
+RpcServer::RpcServer(std::shared_ptr<transport::UTransport> transport,
+                     const v1::UUri& method,
+                     std::optional<v1::UPayloadFormat> format,
+                     std::optional<std::chrono::milliseconds> ttl)
+    : transport_(std::move(transport)), ttl_(ttl) {
+	auto [valid, reason] = Validator::uri::isValidRpcMethod(method);
+	if (!valid) {
+		throw Validator::uri::InvalidUUri(
+		    "Method URI is not a valid URI |  " +
+		    std::string(Validator::uri::message(reason.value())));
+	}
+
+	if (format.has_value()) {
+		if (!UPayloadFormat_IsValid(format.value())) {
+			throw std::out_of_range("Invalid payload format");
+		} else {
+			expected_payload_format_ = std::move(format);
+		}
+	}
+}
+
+RpcServer::ServerOrStatus RpcServer::create(
+    std::shared_ptr<transport::UTransport> transport,
+    const v1::UUri& method_name, RpcCallback&& callback,
+    std::optional<v1::UPayloadFormat> payload_format,
+    std::optional<std::chrono::milliseconds> ttl) {
+	std::unique_ptr<RpcServer> server;
+	try {
+		server.reset(new RpcServer(std::move(transport), method_name,
+		                           std::move(payload_format), ttl));
+	} catch (const Validator::uri::InvalidUUri& e) {
+		v1::UStatus status;
+		status.set_code(v1::UCode::INVALID_ARGUMENT);
+		status.set_message(e.what());
+		return uprotocol::utils::Unexpected(status);
+	} catch (const std::out_of_range& e) {
+		v1::UStatus status;
+		status.set_code(v1::UCode::OUT_OF_RANGE);
+		status.set_message(e.what());
+		return uprotocol::utils::Unexpected(status);
+	} catch (const std::exception& e) {
+		v1::UStatus status;
+		status.set_code(v1::UCode::UNKNOWN);
+		status.set_message(e.what());
+		return uprotocol::utils::Unexpected(status);
+	}
+
+	auto status = server->connect(method_name, std::move(callback));
+	if (status.code() == v1::UCode::OK) {
+		return server;
+	} else {
+		return uprotocol::utils::Unexpected(std::move(status));
+	}
+}
+
+v1::UStatus RpcServer::connect(const v1::UUri& method, RpcCallback&& callback) {
+	callback_ = std::move(callback);
+	auto result = transport_->registerListener(
+	    /*sink_filter=*/{method},
+	    /*listener=*/[this](const v1::UMessage& message) {
+		    // Extract payload data and format from the callback
+		    auto [valid, reason] =
+		        Validator::message::isValidRpcRequest(message);
+		    if (!valid) {
+			    auto response =
+			        datamodel::builder::UMessageBuilder::response(message)
+			            .withCommStatus(v1::UCode::INVALID_ARGUMENT)
+			            .build({std::string(Validator::message::message(
+			                        reason.value())),
+			                    v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT});
```

Comment:
Minor: If the builder is declared before this block somewhere, this could be cleaned up like so:
```suggestion
			    auto message = static_cast<std::string>(Validator::message::message(reason.value()));
			    auto response = builder.withCommStatus(v1::UCode::INVALID_ARGUMENT)
			            .build({std::move(message), v1::UPayloadFormat::UPAYLOAD_FORMAT_TEXT});
```